deployment/aws-python-storage-stack-automation/storage_stack_lifecycle.yaml

```diff
@@ -352,6 +352,202 @@ Resources:
+              - sts:AssumeRole
+      RoleName: !Join ["-",[!Ref AWS::StackName, !Ref AWS::Region,"implodestorage"]]
+  implodelambdarolepolicy:
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyName: !Ref implodelambdarole
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action:
+              - "logs:DescribeLogGroups"
+              - "logs:DeleteLogGroup"
+              - "logs:CreateLogGroup"
+              - "logs:CreateLogStream"
+              - "logs:PutLogEvents"
+              - "logs:PutRetentionPolicy"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "iam:DeletePolicy"
+              - "iam:PassRole"
+              - "iam:DetachRolePolicy"
+              - "iam:GetRole"
+              - "iam:GetRolePolicy"
+              - "iam:DeleteRole"
+              - "iam:DeleteRolePolicy"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "cloudformation:ListExports"
+              - "cloudformation:ListStackInstances"
+              - "cloudformation:DescribeStackResource"
+              - "cloudformation:ListStackResources"
+              - "cloudformation:DescribeStackEvents"
+              - "cloudformation:DescribeStackInstance"
+              - "cloudformation:GetTemplateSummary"
+              - "cloudformation:DescribeStacks"
+              - "cloudformation:GetTemplate"
+              - "cloudformation:DeleteStack"
+              - "cloudformation:ListStacks"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "sqs:ListQueues"
+              - "sqs:GetQueueUrl"
+              - "sqs:ListQueueTags"
+              - "sqs:ReceiveMessage"
+              - "sqs:DeleteMessage"
+              - "sqs:RemovePermission"
+              - "sqs:GetQueueAttributes"
+              - "sqs:ListDeadLetterSourceQueues"
+              - "sqs:DeleteQueue"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "s3:*"
+              - "sns:Unsubscribe"
+              - "sns:SetTopicAttributes"
+              - "sns:GetTopicAttributes"
+              - "sns:RemovePermission"
+              - "sns:DeleteTopic"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "lambda:DeleteFunction"
+              - "lambda:InvokeFunction"
+              - "lambda:GetEventSourceMapping"
+              - "lambda:GetFunction"
+              - "lambda:RemovePermission"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "secretsmanager:GetSecretValue"  
+            Resource: !Ref wsapi
+          - Effect: Allow
+            Action:
+              - "sts:AssumeRole"
+
+            Resource: !GetAtt
+              - implodestoragelambda
+              - Arn
+      Roles:
+        - !Ref implodelambdarole
+  implodestoragelambda:
```

Comment:
What do you mean by `implode`?

Reply:
just to tear down the resources fss created for that bucket. We can rename.

Reply:
How about `delete`, `remove` or `teardown`?

Reply:
No problem. I changed it over to 'remove'.